src/mip/coefficient_strengthening.jl

```diff
@@ -0,0 +1,176 @@
+"""
+Perform coefficient strengthening on each row and on each integer variables
+
+In particular, for the i-th constraints of the form
+aᵢ x ⩽ bᵢ and let xⱼ be a integer/binary variable
+If aᵢⱼ ⩾ d = bᵢ - Mᵢⱼ - aᵢⱼ (uⱼ - 1) > 0
+where Mᵢⱼ is maximal activity of the i-th row without considering xⱼ
+and uⱼ is upper bound of xⱼ, then
+aᵢⱼ ≔ aᵢⱼ - d
+bᵢ ≔ bᵢ - duⱼ
+Also perform coefficient strengthening on constraints of the form aᵢ x ⩾ cᵢ
+with similar update rule
+
+Skip ranged constraints, i.e, cᵢ ⩽ aᵢ x ⩽ bᵢ
+
+The new coefficients are stored by updating ps.pb0
+If some coefficient is reduced to 0, it is kept as 0
+in both ps.pb0.arows and ps.pb0.acols, and only the nonzeros are updated
+
+If a row is reduced to 0 after this step, it is set to be inactive
+If a column is reduced to 0 after this step, it is set to be inactive
+
+"""
+function maximal_activity(row::Row{T}, ucol::Vector{T}, lcol::Vector{T})::T where {T}
+    sup = zero(T)
+
+    for (j, aij) in zip(row.nzind, row.nzval)
+        if aij > zero(T)
+            sup += aij*ucol[j]
+        elseif aij < zero(T)
+            sup += aij*lcol[j]
+        end
+    end
+    return T(sup)
+end
+
+function minimal_activity(row::Row{T}, ucol::Vector{T}, lcol::Vector{T})::T where {T}
+    inf = zero(T)
+
+    for (j, aij) in zip(row.nzind, row.nzval)
+        if aij > zero(T)
+            inf += aij*lcol[j]
+        elseif aij < zero(T)
+            inf += aij*ucol[j]
+        end
+    end
+    return T(inf)
+end
+
+function upperbound_strengthening!(ps::PresolveData{T}, i::Int, j_index::Int, j::Int, max_act) where {T}
```

Comment:
A short docstring would be useful; having both `j` and `j_index` as arguments is a bit confusing.